dataset/src/main/scala/frameless/TypedColumn.scala

```diff
@@ -209,11 +209,16 @@ sealed trait TypedAggregate[T, A] extends UntypedExpression[T] {
   def aencoder: TypedEncoder[A]
 }
 
+/** Represents the result of an aggregation function
+  * @tparam T The type of TypedDataset elements
+  * @tparam A Aggregation type
+  * @tparam U Selected type
+  */
 sealed class TypedAggregateAndColumn[T, A, U](expr: Expression)(
   implicit
   val aencoder: TypedEncoder[A],
   uencoder: TypedEncoder[U]
-) extends TypedColumn[T, U](expr) with TypedAggregate[T, A] {
+) extends TypedColumn[T, A](expr) with TypedAggregate[T, A] {
```

Comment:
For instance, when we do `select(sum(xxx))` and `groupBy(yyy).agg(sum(xxx))` where `xxx` is `Long`, in `groupBy` we get `Long`, and in `select` we get `Option[Long]`